src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java

```diff
@@ -72,10 +74,43 @@ public PartAlignmentOffset findOffsets(Part part, BoardLocation boardLocation,
 
         // Pre-rotate to minimize runout
         double preRotateAngle = 0;
-        if (preRotate && boardLocation != null && placementLocation != null) {
-            preRotateAngle =
-                    Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation)
-                           .getRotation();
+        if (preRotate) {
+			if(part == null || nozzle.getPart() == null) { throw new Exception("no part on nozzle"); }			
+			if((!part.toString().equals(nozzle.getPart().toString()))) { throw new Exception("Part mismatch with part on nozzle"); }
+			part=nozzle.getPart();
+			double angle = placementLocation.getRotation();
+			if(boardLocation!=null) {
+				angle = Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation).getRotation();              
+			}
+			angle=angleNorm(angle,180.);
+			double placementAngle=angle;
+                        MovableUtils.moveToLocationAtSafeZ(nozzle, VisionUtils.getBottomVisionCamera().getLocation().add(new Location(part.getHeight().getUnits(), 0.0, 0.0,part.getHeight().getValue(), 0.0)).derive(null, null, null, angle));
+                        CvPipeline pipeline = partSettings.getPipeline();
+                        pipeline.setCamera(VisionUtils.getBottomVisionCamera());
+                        pipeline.setNozzle(nozzle);
+			pipeline.process();
+			if (!((pipeline.getResult("result")).model instanceof RotatedRect)) {
+				throw new Exception("Bottom vision alignment failed for part " + part.getId()
+						+ " on nozzle " + nozzle.getName() + ". No result found.");
+			}
+
+			RotatedRect rect = ((RotatedRect)(pipeline.getResult("result")).model);
+			angle = angleNorm(angleNorm(angle) + angleNorm((rect.size.width < rect.size.height) ? 90 + rect.angle : rect.angle));
+		// error is -angle
+			if(Math.abs(angle)>0.0765) { angle+=0.0567 * Math.signum(angle); } // rounding 
```

Comment:
What does the value 0.0765 represent? I see that 0.0567 is 360/200/16 as in the last PR, but I am not sure what 0.0765 is.

Reply:
Did you ever get an answer for what 0.0765 was?